### PR TITLE
Issue #SB-5529 fix: view course status for mentors fix

### PR DESCRIPTION
**Pull request by raghuv45**

### CODE REVIEW

File: src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts [hunk 1]
```diff
@@ -148,16 +148,22 @@ export class CourseProgressComponent implements OnInit, OnDestroy {
...
-    takeUntil(this.unsubscribe))
-    .subscribe(
-      (apiResponse: ServerResponse) => {
-        this.batchlist = apiResponse.result.response.content;
+    const searchParamsMentor = {
+      courseId: this.courseId,
+      status: ['0', '1', '2'],
+      mentors: [this.userId]
+    };
+    combineLatest(
```
Comment (anoophm) [inline src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts]: please add test case for this

File: src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts [hunk 2]
```diff
@@ -90,25 +91,44 @@ export class BatchDetailsComponent implements OnInit, OnDestroy {
       offset: 0,
       sort_by: { createdDate: 'desc' }
     };
+    const searchParamsCreator =  _.cloneDeep(searchParams);
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts]: please add testcase for this

File: src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts [hunk 3]
```diff
@@ -90,25 +91,44 @@ export class BatchDetailsComponent implements OnInit, OnDestroy {
...
-        } else {
-          this.showBatchList = true;
-        }
-      },
+      searchParamsCreator.filters.createdBy = this.userService.userid;
+      searchParamsMentor.filters.mentors = [this.userService.userid];
+      combineLatest(
+        this.courseBatchService.getAllBatchDetails(searchParamsCreator),
+        this.courseBatchService.getAllBatchDetails(searchParamsMentor),
+      ).pipe(map(results => ({ batchesOwned: results[0].result.response.content, batchesMentored: results[1].result.response.content })))
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts]: add takeUntil to handle observable

File: src/app/client/src/app/modules/learn/components/batch/update-course-batch/update-course-batch.component.ts [hunk 4]
```diff
@@ -178,6 +180,10 @@ export class UpdateCourseBatchComponent implements OnInit, OnDestroy {
         this.disableSubmitBtn = true;
       }
     });
+    if (this.batchDetails.createdBy !== this.userService.userid) {
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/batch/update-course-batch/update-course-batch.component.ts]: add test case

File: src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 
 import { combineLatest,  Subscription ,  Observable ,  Subject } from 'rxjs';
```
Comment (anoophm) [inline src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts]: Travis failed. please check test case before creating pr.

File: src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts [hunk 6]
```diff
@@ -148,16 +148,22 @@ export class CourseProgressComponent implements OnInit, OnDestroy {
...
-        this.batchlist = apiResponse.result.response.content;
+    const searchParamsMentor = {
+      courseId: this.courseId,
+      status: ['0', '1', '2'],
+      mentors: [this.userId]
+    };
+    combineLatest(
+      this.courseProgressService.getBatches(searchParamsCreator),
+      this.courseProgressService.getBatches(searchParamsMentor),
+    ).pipe(map(results => ({ batchesOwned: results[0].result.response.content, batchesMentored: results[1].result.response.content })))
```
[the remark this answers is not in the document]
Comment (raghuv45) [reply src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts]: done

File: src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts (hunk 3 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/learn/components/batch/batch-details/batch-details.component.ts]: done

File: src/app/client/src/app/modules/learn/components/batch/update-course-batch/update-course-batch.component.ts (hunk 4 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/learn/components/batch/update-course-batch/update-course-batch.component.ts]: done

File: src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts (hunk 5 above)
Comment (raghuv45) [reply src/app/client/src/app/modules/dashboard/components/course-progress/course-progress.component.ts]: done

**Pull request merged**